index.js

```diff
@@ -134,7 +134,12 @@ KinesisStream.prototype._write = function (chunk, encoding, done) {
   }
 
   try {
-    var msg = chunk.toString();
+    var msg = chunk;
+
+    if (Buffer.isBuffer(msg) || typeof msg === 'string') {
```

Comment:
If msg is a buffer, i think that toString method should include encoding.
`if (Buffer.isBuffer(msg)) {
  msg = JSON.parse(msg.toString(encoding));
} else if (typeof msg === 'string') {
 msg = JSON.parse(msg);
} `